application/src/main/java/org/togetherjava/tjbot/commands/tags/TagCommand.java

```diff
@@ -43,9 +45,11 @@ public void onSlashCommand(@NotNull SlashCommandEvent event) {
             return;
         }
 
-        event
-            .replyEmbeds(MessageUtils.generateEmbed(null, tagSystem.getTag(id).orElseThrow(),
-                    event.getUser(), TagSystem.AMBIENT_COLOR))
-            .queue();
+        event.replyEmbeds(tagEmbed(id, event.getUser(), event.getCommandString())).queue();
+    }
+
+    private MessageEmbed tagEmbed(String id, User user, String commandString) {
```

Comment:
* bad method name. Rename to what it actually does, a verb like `createTagMessage` or `createTagEmbed` or just `createEmbed`/`createMessage`.
* Missing `@NotNull`
* Instead of taking the user and commandString individually, take the `event` and then extract the user and command string inside the helper method. That makes the call-site (line 48) even easier to read and gives the helper method more freedom as well.